packages/cozy-dataproxy-lib/src/dataproxy/DataProxyProvider.jsx

```diff
@@ -117,21 +118,69 @@ export const DataProxyProvider = React.memo(({ children, options = {} }) => {
     setDataProxy(() => remote)
   }
 
-  const search = async search => {
-    log.log('Send search query to DataProxy')
-    const result = await dataProxy.search(search, options)
-
-    return result
+  const onReceiveMessage = event => {
+    if (!event.origin.includes('dataproxy')) {
+      return
+    }
+    const eventData = event?.data
+    if (eventData && typeof eventData === 'object') {
+      if (
+        eventData.type === 'DATAPROXYMESSAGE' &&
+        eventData.payload === 'READY'
+      ) {
+        onIframeLoaded()
+      }
+    }
   }
 
-  const value = {
-    dataProxyServicesAvailable,
-    search
-  }
+  useEffect(function () {
+    window.addEventListener('message', onReceiveMessage)
+    return function () {
+      window.removeEventListener('message', onReceiveMessage)
+    }
+  })
+
+  useEffect(() => {
+    const doAsync = async () => {
+      // Make a global search
+      const search = async (search, options) => {
+        log.log('Send search query to DataProxy')
+        const result = await dataProxy.search(search, options)
+        return result
+      }
+
+      // Request through cozy-client
+      const requestLink = async (operation, options) => {
+        log.log('Send request to DataProxy : ', operation)
+        if (options.fetchPolicy) {
+          // Functions cannot be serialized and thus passed to the iframe
+          delete options.fetchPolicy
```

Comment:
nit: in the future we may want to recreate a new options object without the fetchPolicy instead of deleting the attribute. This would prevent unexpected side effects.